lib/engine-addon.js

```diff
@@ -126,9 +129,17 @@ var buildVendorCSSTree = memoize(function buildVendorCSSTree(vendorTree) {
 
 var buildEngineJSTree = memoize(function buildEngineJSTree() {
   var engineSourceTree;
-  var treePath = path.resolve(this.root, this.treePaths['addon']);
-  if (existsSync(treePath)) {
+  var treePath;
+  var addonTree = this.options.trees.addon;
```

Comment:
Is it not possible for us to just override `treePaths` directly via a merge or something?